criu/mount.c

```diff
@@ -1851,10 +1851,15 @@ static int restore_shared_options(struct mount_info *mi, bool private, bool shar
 			mi->mnt_id, mi->mountpoint, private, shared, slave);
 
 	if (mi->flags & MS_UNBINDABLE) {
-		if (shared || slave)
+		if (shared || slave) {
 			pr_warn("%s has both unbindable and sharing, ignoring unbindable\n", mi->mountpoint);
-		else
+		} else {
+			if (!mnt_is_overmounted(mi)) {
+				pr_debug("Temporary leave unbindable mount %s as private\n", mi->mountpoint);
```

Comment:
need a comment which will explain why we need to do this. And are you sure that this will not break the restore of shared mounts?

Reply:
> And are you sure that this will not break the restore of shared mounts?

Unbindable mounts don't have shared_id or master_id set, so when binding from it we will treat exactly the same as we do for a private mount, and I set it to be private, everything should be fine. We already have similar fix in vz7-criu for 2 months.